crates/runix/src/flake_ref/mod.rs

```diff
@@ -213,16 +214,25 @@ impl FlakeRef {
     }
 
     /// Parses a URI into a flake reference given the URI and the path to the `parser-util` binary
-    pub fn from_url<T, P>(uri: T, bin_path: P) -> Result<Self, UrlParseError>
+    pub fn from_url<U, P>(uri: U, bin_path: P) -> Result<Self, UrlParseError>
     where
-        T: AsRef<str>,
+        U: AsRef<str> + Clone,
         P: AsRef<Path>,
     {
         let parsed = url_parser::installable_flake_ref(uri, bin_path)?;
         let parsed_ref = parsed.r#ref;
+        Self::from_parsed(&parsed_ref)
+    }
+
+    /// Converts a parsed flake reference from `parser-util` to a [FlakeRef]
+    ///
+    /// This method is agnostic over the resolution level of the parsed flake reference
+    /// e.g. the original flake reference, the resolved flake reference, or the locked
+    /// flake reference.
+    pub fn from_parsed(parsed_ref: &ParsedFlakeReference) -> Result<Self, UrlParseError> {
```

Comment:
Why does this take a borrowed `ParsedFlakeReference`?

Reply:
Because it doesn't need to take ownership of the `ParsedFlakeReference`?